ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+
+    pFlushRecList->SizeOfBlockToBeSaved = pBD_XP->BytesPerRecord;
+    pFlushRecList->pBlockToBeSaved = (void *)pszRecordOnCourse;
+
+    // Test lenght
+    if (MMDetectAndFixDBFWidthChange(hMiraMonLayer, hMMFeature,
+                                     &hMiraMonLayer->MMPolygon.MMAdmDB,
+                                     pFlushRecList, nNumPrivateMMField, 0, 0))
+        return MM_FATAL_ERROR_WRITING_FEATURES;
+
+    // Reassign the point because the function can realloc it.
+    pszRecordOnCourse = hMiraMonLayer->MMPolygon.MMAdmDB.szRecordOnCourse;
+    pFlushRecList->pBlockToBeSaved = (void *)pszRecordOnCourse;
+
+    // Now lenght is sure, write
+    memset(pszRecordOnCourse, 0, pBD_XP->BytesPerRecord);
+    if (MMWriteValueToRecordDBXP(hMiraMonLayer, pszRecordOnCourse,
+                                 pBD_XP->pField, &nElemCount, TRUE))
+        return MM_FATAL_ERROR_WRITING_FEATURES;
+
+    if (!hMMFeature)
+    {
+        if (MMAppendBlockToBuffer(pFlushRecList))
+            return MM_FATAL_ERROR_WRITING_FEATURES;
+        hMiraMonLayer->MMPolygon.MMAdmDB.pMMBDXP->nRecords++;
+        return MM_CONTINUE_WRITING_FEATURES;
+    }
+
+    MMWriteValueToRecordDBXP(hMiraMonLayer, pszRecordOnCourse,
+                             pBD_XP->pField + 1, &nVerticesCount, TRUE);
+
+    MMWriteValueToRecordDBXP(hMiraMonLayer, pszRecordOnCourse,
+                             pBD_XP->pField + 2, &pPolHeader->dfPerimeter,
+                             FALSE);
+
+    MMWriteValueToRecordDBXP(hMiraMonLayer, pszRecordOnCourse,
+                             pBD_XP->pField + 3, &pPolHeader->dfArea, FALSE);
+
+    MMWriteValueToRecordDBXP(hMiraMonLayer, pszRecordOnCourse,
+                             pBD_XP->pField + 4, &pPolHeader->nArcsCount, TRUE);
+
+    MMWriteValueToRecordDBXP(hMiraMonLayer, pszRecordOnCourse,
+                             pBD_XP->pField + 5, &pPolHeader->nRingsCount,
+                             TRUE);
+
+    pFlushRecList->SizeOfBlockToBeSaved = pBD_XP->BytesPerRecord;
+    if (MMAddFeatureRecordToMMDB(
+            hMiraMonLayer, hMMFeature, &hMiraMonLayer->MMPolygon.MMAdmDB,
+            pszRecordOnCourse, pFlushRecList,
+            &hMiraMonLayer->MMPolygon.MMAdmDB.pMMBDXP->nRecords,
+            nNumPrivateMMField))
+        return MM_FATAL_ERROR_WRITING_FEATURES;
+    return MM_CONTINUE_WRITING_FEATURES;
+}
+
+// Close the MiraMon database associated with a vector layer.
+static int MMCloseMMBD_XPFile(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                              struct MMAdmDatabase *MMAdmDB)
+{
+    if (!hMiraMonLayer)
+        return 1;
+
+    if (hMiraMonLayer->ReadOrWrite == MM_WRITTING_MODE)
+    {
+        if (!MMAdmDB->pFExtDBF)
+        {
+            // In case of 0 elements created we have to
+            // create an empty DBF
+            if (hMiraMonLayer->bIsPolygon)
+            {
+                if (hMiraMonLayer->TopHeader.nElemCount <= 1)
```

Comment:
is <= 1 correct here ? Just a few lines below the test is == 0

Reply:
Let me think about it cause exists the concept of universal polygon (or polygon zero) that only exists if there is another polygon on the layer. So it's clearer to talk about to <=1 cause == 0 could be and error.
So, I leave it open to think about it later.

Reply:
<=1 it's ok because in case of 1 element (the universal polygon) this layers will be considered as empty